File: microon-services/microon-services-auth/src/main/scala/microon/services/auth/AuthService.scala
package microon.services.auth

import java.util.concurrent.Future
import microon.spi.scala.activeobject.Void

trait AuthService {

  def auth(authRequest: AuthRequest): Future[Void]

  def isLoggedIn(userId: String): Future[Boolean]

  def lastLogInFailureMessage(userId: String): Future[Option[String]]

  def logout(userId: String): Future[Void]

}

File: microon-services/microon-services-auth/src/main/java/microon/services/auth/api/java/AuthService.java
package microon.services.auth.api.java;

import microon.spi.scala.activeobject.Void;

import java.util.concurrent.Future;

public interface AuthService {

    Future<Void> auth(AuthRequest authRequest);

    Future<Boolean> isLoggedIn(String userId);

    Future<String> lastLogInFailureMessage(String userId);

    Future<Void> logout(String userId);

}